File: ios/Anemone SDK/general/client/HttpResponse.swift
//
//  HttpResponse.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 21/09/2017.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public struct HttpResponse {
    public let request: HttpRequest
    public let headers: [String: String]
    public let code: Int
    public let body: Data

    public var isSuccessful: Bool {
        code >= 200 && code < 400
    }

    /**
     * Remember that you need to send <HttpRequest.includeTotalHeader(Key/Value)>
     * in the headers request to receive the total count.
     * You have a few examples:
     * <AnemoneSDK.MatchService.countMatches>
     * <AnemoneSDK.AcademyClassService.countCourses>
     * <AnemoneSDK.ActivityService.countTournaments>
     */
    public var totalCount: Int? {
        headers["total"].let { Int($0) }
    }
}

extension HttpResponse: Equatable { }
public func == (lhs: HttpResponse, rhs: HttpResponse) -> Bool {
    lhs.request == rhs.request &&
        lhs.headers == rhs.headers &&
        lhs.code == rhs.code &&
        lhs.body == rhs.body
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/general/client/HttpResponse.kt
package com.anemonesdk.general.client

import com.playtomic.foundation.extension.tryOrNull

/**
 * Created by agarcia on 25/09/2017.
 */
data class HttpResponse(
    val request: HttpRequest,
    val headers: Map<String, String>,
    val code: Int,
    val body: ByteArray
) {

    val isSuccessful: Boolean = code in 200 until 400

    /**
     * Remember that you need to send HttpRequest.includeTotalHeader(Key/Value)
     * in the headers request to receive the total count.
     * You have a few examples:
     * @see com.anemonesdk.service.MatchService.countMatches
     * @see com.anemonesdk.service.AcademyClassService.countCourses
     * @see com.anemonesdk.service.ActivityService.countTournaments
     */
    val totalCount: Int?
        get() = headers["total"]?.let { tryOrNull { it.toInt() } }
}
